packages/core/src/components/slides/slides.tsx

```diff
@@ -78,188 +84,58 @@ export class Slides {
-
-
-  /**
-   * @input {boolean} If true, allows you to use "parallaxed" elements inside of
-   * slider.
-   */
-  @Prop() parallax: boolean = false;
-
-  /**
-   * @input {number} Slides per view. Slides visible at the same time. Default: `1`.
+   * Options to pass to the swiper instance.
+   * See http://idangero.us/swiper/api/ for valid options
    */
-  @Prop() slidesPerView: number | 'auto' = 1;
+  @Prop() opts: any;
 
   /**
-   * @input {number} Distance between slides in px. Default: `0`.
+   * Show or hide the pager
    */
-  @Prop() spaceBetween: number = 0;
-
-  /**
-   * @input {number} Duration of transition between slides
-   * (in milliseconds). Default: `300`.
-   */
-  @Prop() speed: number = 300;
-
-
-  /**
-   * @input {boolean} If true, enables zooming functionality.
-   */
-  @Prop() zoom: boolean;
-
-  /**
-   * @input {boolean} If true, enables keyboard control
-   */
-  @Prop() keyboardControl: boolean;
-
+  @Prop() pager: boolean = true;
 
   render() {
     return (
       <div class='swiper-container' data-dir='rtl'>
         <div class='swiper-wrapper'>
-          <slot></slot>
+          <slot />
         </div>
-        <div class={{
-          'swiper-pagination': true,
-          'hide': !this.pager
-        }}></div>
+        <div
+          class={{
+            'swiper-pagination': true,
+            hide: !this.pager
+          }}
+        />
       </div>
     );
   }
 
-  /**
-   * @hidden
-   * Height of container.
-   */
-  private height: number;
-
-  /**
-   * @hidden
-   * Width of container.
-   */
-  private width: number;
-
-  /**
-   * @hidden
-   * Enabled this option and swiper will be operated as usual except it will
-   * not move, real translate values on wrapper will not be set. Useful when
-   * you may need to create custom slide transition.
-   */
-  private virtualTranslate = false;
-
-  /**
-   * @hidden
-   * Set to true to round values of slides width and height to prevent blurry
-   * texts on usual resolution screens (if you have such)
-   */
-  private roundLengths = false;
-
-  // Slides grid
-
-  /**
-   * @hidden
-   */
-  private originalEvent: any;
-
-  /**
-   * Private properties only useful to this class.
-   * ------------------------------------
-   */
-  private _init: boolean;
-  private _tmr: number;
-
-  /**
-   * Properties that are exposed publicly but no docs.
-   * ------------------------------------
-   */
-  /** @hidden */
-  private container: HTMLElement;
-  /** @hidden */
-  private slidesId: number;
-  /** @hidden */
-  private slideId: string;
-
-
-  constructor(
-  ) {
+  constructor() {
     this.slidesId = ++slidesId;
     this.slideId = 'slides-' + this.slidesId;
   }
 
-  private _initSlides() {
-    if (!this._init) {
+  private initSlides() {
+    if (!this.init) {
       console.debug(`ion-slides, init`);
 
       this.container = this.el.children[0] as HTMLElement;
 
+      // Base options, can be changed
       var swiperOptions = {
```

Comment:
I don't know how it's handled in the Ionic project, but I think `let` and `const` should be preferred over `var`.